lib/block_reservation.rb

```diff
@@ -0,0 +1,14 @@
+module Hotel
+  class BlockReservation
+    attr_accessor :block_id, :reservation_id, :room, :start_date, :end_date, :price_per_night
+
+    def initialize(block_id:,  reservation_id:, room:, start_date:, end_date:, price_per_night:)
```

Comment:
The only difference between this and a regular reservation is that a block reservation has a block ID. One option that might simplify your design is to consolidate both in the `Reservation` class, and add in an optional block ID, which would be `nil` if this reservation isn't part of a block.